File: RobotArm/InputFiles/Driving/Ref/Examples/trans1.m
function y = trans1(t, coordNum, pos, vel, load_pos, load_vel, top_pos, top_vel)
% t: time
% coordNum: index of a point you are currently targeting
% pos: measured positions of actuators
% vel: measured velocities of actuators
% load_pos: measured load position
% load_vel: measured load velocity
% top_pos: measured position of the top of the crane
% top_vel: measured velocity of the top of the crane
% y: referent valve opening [m]


% bufferTrans1 is the global buffer used for this function.
% Example: a = bufferTrans1[4] assigns 4th element of the list to variable a


% We recommend you to hardcode all configuration points you have to go through 
% (points you got in inverse kinematics).
% Example:
% actuatorRefOne = [1, 2, 3, 4, 5];
% actuatorRefTwo = [1, 2, 3, 4, 5];
% actuatorRefThree = [1, 2, 3, 4, 5];
% actuatorRefVector = [actuatorRefThree; 
% 		       actuatorRefOne; 
% 		       actuatorRefThree; 
% 		       actuatorRefTwo; 
% 		       actuatorRefThree];
	y = 0;
end

File: RobotArm/InputFiles/Driving/Regulators/trans1.c
#include "c_code.h"


// t: time
// coord_num: index of a point you are currently targeting
// pos: measured positions of actuators
// vel: measured velocities of actuators
// load_pos: measured load position
// load_vel: measured load velocity
// top_pos: measured position of the top of the crane
// top_vel: measured velocity of the top of the crane
// returns: referent valve opening [m]
inline double trans1(double t, int coord_num, Measurement pos, Measurement vel, 
           Coords load_pos, Coords load_vel,
           Coords top_pos ,Coords top_vel)
{
    // buffer_trans1 is the global buffer used for this function.
    // Example: a = buffer_trans1[4] assigns 5th element of the list to variable a


    // We recommend you to hardcode all configuration points you have to go through 
    // (points you got in inverse kinematics).
    // Example:
    // double actuator_ref_one[5] = {0, 1, 2, 3, 4};
    // double actuator_ref_two[5] = {0, 1, 2, 3, 4};
    // double actuatorr_ref_three[5] = {0, 1, 2, 3, 4};
    // double actuator_ref_array[5] = {actuatorr_ref_three,
    //                                 actuator_ref_one,
    //                                 actuatorr_ref_three,
    //                                 actuator_ref_two,
    //                                 actuatorr_ref_three};

    double a = buffer_trans1[4] = 5;
    buffer_trans1[4]++;
    return a;
}